crates/sdk/src/tx.rs

```diff
@@ -480,14 +480,38 @@ pub fn display_batch_resp(context: &impl Namada, resp: &TxResponse) {
         false
     };
     let batch_results = resp.batch_result();
-    if !batch_results.is_empty() {
+    // If the batch contains at least one transaction
+    if let Some((first_inner_hash, first_result)) = batch_results.first() {
         if !wrapper_successful {
-            display_line!(
-                context.io(),
-                "Since the batch in its entirety failed, none of the \
-                 transactions listed below have been committed. Their results \
-                 are provided for completeness.",
-            );
+            // Check if fees were paid via the shielded pool, in this case the
+            // first transaction of the batch was committed regardless of the
+            // batch failure (even for atomic batches)
+            let masp_fee_payment = if let InnerTxResult::Success(res) =
+                first_result
+            {
+                res.events.iter().any(|event| {
+                    event.kind() == &namada_tx::event::masp_types::FEE_PAYMENT
+                })
+            } else {
+                false
+            };
```

Comment:
suggestion: this is more compact and imo more readable
```suggestion
            let masp_fee_payment = matches!(
                first_result,
                InnerTxResult::Success(res)
                    if res.events.iter().any(|event| {
                        event.kind() == &namada_tx::event::masp_types::FEE_PAYMENT
                    }),
            );

```

Reply:
Done in a456b1a3fb36de72304c30ff2408a3be47b50058